grammars/puppet.cson

```diff
@@ -394,23 +256,252 @@
+    'match': '(?:(?:\\D&\\W)|\\s|^|\\[|\\{|,|;|\\()(\\d+([.]\\d+)?)(?!\\w)'
+    'name': 'meta.control.puppet'
+  'define-parameters':
+    'begin': '\\('
+    'beginCaptures':
+      '0':
+        'name': 'punctuation.definition.classparameter.begin.puppet'
+    'end': '\\)'
+    'endCaptures':
+      '0':
+        'name': 'punctuation.definition.classparameter.end.puppet'
+    'name': 'meta.classparameter.language.puppet'
+    'patterns': [
+      {
+        'include': '#parameter-default-types'
+      }
+    ]
+
+  'definitions':
+    'begin': '^\\s*(node|define|class|application)\\s+((?:[-_A-Za-z0-9"\'.]+::)*[-_A-Za-z0-9"\'.]+)\\s*'
+    'beginCaptures':
+      '1':
+        'name': 'storage.type.puppet'
+      '2':
+        'name': 'entity.name.type.class.puppet'
+    'end': '{'
+    'endCaptures':
+      '0':
+        'name': 'punctuation.definition.class.begin.puppet'
+    'name': 'meta.definition.class.puppet'
+    'patterns': [
+      {
+        'begin': '\\b(inherits)\\b\\s+'
+        'beginCaptures':
+          '1':
+            'name': 'storage.modifier.puppet'
+        'end': '{'
+        'endCaptures':
+
+            'name': 'punctuation.definition.class.begin.puppet'
+        'name': 'meta.definition.class.inherits.puppet'
+        'patterns': [
+          {
+            'match': '\\b((?:[-_A-Za-z0-9".]+::)*[-_A-Za-z0-9".]+)\\b'
+            'name': 'entity.other.inherited-class.puppet'
+          }
+        ]
+      }
+      {
+        'include': '#define-parameters'
+      }
+    ]
+  'resources':
+    'begin': '((?:[-_A-Za-z0-9".]+::)*[-_A-Za-z0-9".]+)\\s*{'
+    'beginCaptures':
+      '1':
+        'name': 'storage.type.puppet'
+    'name': 'meta.definition.resource.puppet'
+    'end': '}'
+    'patterns': [
+      {
+        'match': ':'
+        'captures':
+          '0':
+            'name': 'punctuation.classtitle.puppet'
+        'name': 'meta.title.puppet'
+      }
+      {
+        'include': '#line_comment'
+      }
+      {
+        'include': '#strings'
+      }
+      {
+        'include': '#variables'
+      }
+      {
+        'include': '#metaparameters'
+      }
+      {
+        'beginCaptures':
+          '1':
+            'name': 'name.parameter.resource.puppet'
+          '2':
+            'name': 'punctuation.separator.key-value.puppet'
+        'begin': '(\\w+)\\s*(=>)'
+        'name': 'meta.parameter.resource.puppet'
+        'end': ',|\\n|;'
+        'patterns': [
+          {
+            'include': '#parameter-default-types'
+          }
+        ]
+      }
+    ]
+  'puppet':
+    'patterns': [
+      {
+        'include': '#parameter-default-types'
+      }
+      {
+        'include': '#resources'
+      }
+      {
+        'include': '#conditionals'
+      }
+      {
+        'match': '((\\$?)"?[a-zA-Z_\\x{7f}-\\x{ff}][a-zA-Z0-9_\\x{7f}-\\x{ff}]*"?):(?=\\s+|$)'
+        'name': 'entity.name.section.puppet'
+      }
+      {
+        'begin': '(?i)\\b(import|include)\\b\\s*'
```

Comment:
`\\b` followed by `\\s*` is redundant (for the most part).  `\\b` would probably suffice.